database/src/main/flow_patterns/insert_additional_data.sql

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION flow_patterns.set_additional_data(
+    IN  i_flow_name             TEXT,
+    IN  i_ad_name               TEXT,
+    IN  i_ad_default_value      TEXT,
+    IN  i_by_user               TEXT,
+    OUT status                  INTEGER,
+    OUT status_text             TEXT,
+    OUT id_fp_additional_data   BIGINT,
+    OUT key_fp_flow             BIGINT
+) RETURNS record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: flow_patterns.set_additional_data(4)
+--      Adds an additional data pattern for the given flow.
+--
+-- Parameters:
+--      i_flow_name             - flow name the entry belongs to
+--      i_ad_name               - name of the additional data entry
+--      i_ad_default_value      - default value of the additional data entry
+--      i_by_user               - user initiating the change
+
+--
+-- Returns:
+--      status                  - Status code
+--      status_text             - Status text
+--      id_fp_additional_data   - id of the additional data pattern entry
+--      key_fp_flow             - id of the flow the additional data pattern were added into
+--
+-- Status codes:
+--      11                  - Additional data entry created
+--      41                  - Flow not found
```

Comment:
Is there a reason why this is `41 Flow not found` (as in type `41 - master record not found`), while other calls respond by `40 Flow not found` (as in type `40 - general not found`)?

Reply:
In this case master record is not found in an attempt to add data from master-detail relationship.
But it's totally possible to use 40 as well, if the contact between DB and app would consider in more useful 😉

Reply:
In this case master record is not found in an attempt to add data from master-detail relationship.
But it's totally possible to use 40 as well, if the contact between DB and app would consider in more useful 😉